projects/avodata/AvodataSolverTool.java

```diff
@@ -0,0 +1,142 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Scanner;
+
+import org.apache.commons.math3.analysis.function.Abs;
+import org.apache.commons.math3.linear.Array2DRowRealMatrix;
+import org.apache.commons.math3.linear.DecompositionSolver;
+import org.apache.commons.math3.linear.MatrixUtils;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+import org.apache.commons.math3.linear.SingularValueDecomposition;
+import org.apache.commons.math3.util.CombinatoricsUtils;
+
+/** To build: bazel build //projects/avodata:ls-alg_tool To run: bazel-bin/projects/avodata/ls-alg_tool */
+/** This class reads matrix from csv file, ignores the parmeters and calculate 
+/** using least squares */
+public class AvodataSolverTool {
+
+  static final String fpath = "/projects/avodata/check.csv";
+
+  public static double factorial(int n) {
+    return CombinatoricsUtils.factorialDouble(n);
+  }
+
+  public static double nChooseK(int n, int k) {
+    return CombinatoricsUtils.binomialCoefficientDouble(n, k);
+  }
+
+  public static double[][] generateSyntheticA(int numOfProfessions) {
+
+    int numOfraws = (int) nChooseK(numOfProfessions, 2);
+    RealMatrix matrix = new Array2DRowRealMatrix(numOfraws, numOfProfessions);
+
+    int row = 0;
+    for (int i = 0; i < numOfProfessions; i++)
+      for (int j = i + 1; j < numOfProfessions; j++) {
+        matrix.addToEntry(row, i, 1);
+        matrix.addToEntry(row, j, -1);
+        row++;
+      }
+    return matrix.getData();
+  }
+
+  public static double[][] generateSyntethicC(RealMatrix m, int i, RealMatrix a) {
+    RealMatrix c = a.multiply(m.getRowMatrix(i).transpose());
+    return c.getData();
+  }
+
+  public static RealVector solve(RealMatrix a, RealVector c) {
+
+    SingularValueDecomposition s = new SingularValueDecomposition(a);
```

Comment:
Did you consider using `org.apache.commons.math3.fitting.leastsquares`?
https://commons.apache.org/proper/commons-math/userguide/leastsquares.html

http://commons.apache.org/proper/commons-math/javadocs/api-3.6.1/org/apache/commons/math3/fitting/leastsquares/LeastSquaresProblem.html

BTW, where did you come up with the `SingularValueDecomposition` solution?

Reply:
I didn't very well the example they have there when i found apache common library so i tried to find 
simpler solution, I found the  **SingularValueDecomposition** solution on stack overflow.
I can try use the **org.apache.commons.math3.fitting.leastsquares** if you think its better.

Reply:
Let's try to use the leastsquares again and see where the problem is.

Reply:
Can you link to the StackOverflow you used?

Reply:
Sure

https://stackoverflow.com/questions/5010682/numpy-linalg-lstsq-equivalent-in-apache-commons-math-java-library